.vsts-dotnet-ci.yml

```diff
@@ -4,7 +4,38 @@ trigger:
 - vs*
 
 jobs:
+- job: CheckVersionBumpOnReleaseBranches
+  displayName: "Check Version Bump On Release Branches"
+  steps:
+  - powershell: |
+      $versionsFile = "eng/Versions.props"
+      $changedFiles = git diff --name-only HEAD HEAD~1
+      $changedVersionsFile = $changedFiles | Where-Object { $_ -eq $versionsFile }
+      $isVersionBumped = $false
+      if ($changedVersionsFile -ne $null) {
+        $difference = git diff HEAD~1 $versionsFile
```

Comment:
What happens if this is the initial commit on a new release branch - then it'll probably fail, correct?

We might prevent that by looking for `<DotNetFinalVersionKind>release</DotNetFinalVersionKind>` and if that is not present in the previous commit - than version bump is not required.


See e.g.: https://github.com/dotnet/msbuild/pull/9935/files

Reply:
Yes, it will fail for the initial commit. I will fix it. 
Also we need to consider dependency update on release branch that doesn't need to bump up the version, e.g. https://github.com/dotnet/msbuild/pull/9574/files#diff-1ea18ff65faa2ae6fed570b83747086d0317f5e4bc325064f6c14319a9c4ff67.

Reply:
Thank you!

Btw. the pattern might possibly match even if we move the tag from separate line (e.g. https://github.com/dotnet/msbuild/pull/9661/files) - but that sounds as very acceptable limitation.

I have couple nit picking asks:
* please comment the `$initialCommitPattern` regex - mentioning that 'DotNetFinalVersionKind' is expected to be added only during the initial setup of the branch
* do not exclude the meastro PRs - we need to bump the version on majority of those as well - so it's better to mandate it for all

Reply:
> Btw. the pattern might possibly match even if we move the tag from separate line (e.g. https://github.com/dotnet/msbuild/pull/9661/files) - but that sounds as very acceptable limitation.
Yes, the pattern for checking version bump up matches no matter the tag `DotNetFinalVersionKind` is on the same line or not.

I will resolve others.